File: 1_intro/makeArrayConsecutive2.js
function makeArrayConsecutive2(statues) {
    statues.sort(function(a, b){return a-b})
    
    var checkSpace = 0
    var neededStatues = 0
    
    for (i = 0; i < statues.length; i++) {
        var firstStatue = statues[i],
            nextStatue = statues[i + 1]
        checkSpace =  (nextStatue - firstStatue) - 1
        if (checkSpace > 0) {
            neededStatues = neededStatues + checkSpace
        }
    }
    console.log(statues)
    return neededStatues
}

File: 1_intro/2_edge_of_the_ocean/6_makeArrayConsecutive2/makeArrayConsecutive2.lua
function makeArrayConsecutive2(statues)
  table.sort(statues, function(a, b) 
    return a < b
  end)

  checkSpace = 0
  neededStatues = 0

  for i = 1, table.getn(statues) - 1 do
    firstStatue = statues[i]
    nextStatue = statues[i + 1]
    checkSpace = (nextStatue - firstStatue) - 1
    if checkSpace > 0 then
      neededStatues = neededStatues + checkSpace
    end
  end
  return neededStatues
end